### PR TITLE
Fixes DisconnectedInputError in #1275 and #1386

### PR DESCRIPTION
**Pull request by berleon**
This pull request fixes the DisconnectedInputError in issues #1275 and #1386

### CODE REVIEW

File: keras/layers/normalization.py [hunk 1]
```diff
@@ -57,16 +57,6 @@ def build(self):
...
-        # initialize self.updates: batch mean/std computation
-        X = self.get_input(train=True)
-        m = K.mean(X, axis=0)
-        std = K.mean(K.square(X - m) + self.epsilon, axis=0)
-        std = K.sqrt(std)
-        mean_update = self.momentum * self.running_mean + (1-self.momentum) * m
-        std_update = self.momentum * self.running_std + (1-self.momentum) * std
-        self.updates = [(self.running_mean, mean_update),
-                        (self.running_std, std_update)]
-
```
Comment (fchollet) [inline keras/layers/normalization.py]: These lines belong in `build`. Can we detail why you thought it was necessary to move them to `get_output`?

Comment (berleon) [inline keras/layers/normalization.py]: Well, the input of BatchNormalization Layer can change after the `build` method was called. For example if the layer is included in a nested Sequential / Graph models as described in #1386. 
Having these lines in `build` results in a DisconnectedInputError since the `self.updates` rules use the wrong input.
In short there is no guarantee that `self.get_input` returns the same input in `build` and `get_output`.
